lib/deliver/runner.rb

```diff
@@ -20,8 +20,8 @@ def run
       verify_version if options[:app_version].to_s.length > 0
       upload_metadata
 
-      has_binary = options[:ipa] || options[:pkg]
-      if !options[:skip_binary_upload] && has_binary
+      has_binary = (options[:ipa] || options[:pkg])
+      if !options[:skip_binary_upload] && !options[:build_number] && has_binary
```

Comment:
As I can see here, the user can't provide an `ipa` and a `build_number` (which makes sense), however currently we'll fail silently. 

There is a cool new feature in the fastlane config system you can see here: https://github.com/fastlane/deliver/blob/master/lib/deliver/options.rb#L37-L40

Basically you can define that `skip_binary_upload` doesn't like `ipa` and `pkg`, so that the user will get a nice error message :+1: